metrics/metrics.go

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 GRAIL, Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Package metrics defines a set of primitives for expressing and managing
+// metrics within Bigslice. A metric (such as a counter) is declared by a
+// toplevel registration mechanism (e.g., NewCounter()). All metrics must be
+// declared before a Bigslice session is started.
```

Comment:
you should probably say that metrics are implicitly named by their variable name and that that variable is needed to access the value. This all flows naturally when you see a complete example, but I don't think it's obvious from this comment. A test Example would be nice too.

Reply:
yeah, possibly the way to document this is to describe it as a "handle". i've attempted to clarify this, and also added an example test like you suggested: let me know what you think.